src/common/opencl.h

```diff
@@ -337,13 +337,33 @@ int dt_opencl_get_kernel_work_group_size(const int dev, const int kernel, size_t
 int dt_opencl_set_kernel_arg(const int dev, const int kernel, const int num, const size_t size,
                              const void *arg);
 
+/** wrap opencl arguments */
+#define CLWRAP(size, ptr) (const size_t)0xF111E8, (const size_t)size, (const void *)ptr
```

Comment:
Ok, took me some time to understand this one. It is a marker to be able to parse the varargs in the implementation of `dt_opencl_enqueue_kernel_2d_args()`.

I would use a #define for this marker and add some comments to help future readers.

Reply:
> It is a marker to be able to parse the varargs

It is not really _needed_. It is a magic marker to prevent users from just putting variables directly when calling `dt_opencl_enqueue_kernel_2d_args` and forgetting to wrap them in CLARG or CLLOCAL. They should never use this marker by itself, because that would defeat the purpose of the extra check. If I add a #define for just the marker, that would increase the chance that somebody uses it incorrectly.

In theory, this marker could be used only in the debug version and be stripped out in the release version (so we'd put only 2 values on the stack for each argument instead of 3). But the "speedup" would of course be minimal. And I fear many developers don't use the debug version at all and this is meant to help _them_, so...

Yes, I could put all of this in a comment :-)